x/src/cargo.rs

```diff
@@ -86,19 +88,35 @@ impl Cargo {
 
     pub fn run(&mut self) -> Result<()> {
         self.inner.stdout(Stdio::inherit()).stderr(Stdio::inherit());
-        self.do_run().map(|_| ())
+        self.do_run(true).map(|_| ())
     }
 
     pub fn run_with_output(&mut self) -> Result<Vec<u8>> {
         self.inner.stderr(Stdio::inherit());
-        self.do_run().map(|o| o.stdout)
+        // Since system out hijacked don't log for this command
+        self.do_run(false).map(|o| o.stdout)
     }
 
-    fn do_run(&mut self) -> Result<Output> {
+    fn do_run(&mut self, log: bool) -> Result<Output> {
         if !self.pass_through_args.is_empty() {
             self.inner.arg("--").args(&self.pass_through_args);
         }
+        // once all the arguments are added to the command we can log it.
+        if log {
+            info!("Executing: {:?}", &self.inner);
+        }
```

Comment:
I don't think its worth piping through if we should log or not, remove the bool that you're feeding through and just always log and let the user's set RUST_LOG or a top level command switch that just doesn't enable env_logger

Reply:
The reason is to prevent calls to pkgid from logging, it's of no value to the user.

Reply:
Maybe inspect the subcommand instead?

Reply:
I'm not sure I follow, could you provide a little more context?

Reply:
I'm going to be calling grcov, and not strictly cargo commands soon, so don't want to start parsing the command, at least without capturing all this in the type info first.